NodeProg/Infrastructure/NumberNode.cs

```diff
@@ -21,14 +22,129 @@ public NumberNode(long value)
+                return null;
+            var trimChars = new List<char>();
+            if (allowLeadingWhitespace)
+                trimChars.Add(' ');
+            if (allowTrailingWhitespace)
+                trimChars.Add('\t');
+
+            var digits = new char[] { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9' };
+
+            var position = startIndex;
+            var minusFlag = false;
+            var value = new Int64();
+            for (; position < startIndex + length;)
+            {
+                if (input[position] == '+' || input[position] == '-')
+                {
+                    if (input[position] == '-')
+                        minusFlag = true;
+                    position++;
+                    break;
+                }
+                var trimFlag = false;
+                foreach (var tChar in trimChars)
+                {
+                    if (input[position] == tChar)
+                    {
+                        trimFlag = true;
+                        position++;
+                        break;
+                    }
+                }
+                if (trimFlag)
+                    continue;
+                break;
+            }
+
+            var int64s = new List<int>() { };
+
+            for (; position < startIndex + length;)
+            {
+                var digitFlag = false;
+                foreach (var digit in digits)
+                {
+                    if (input[position] == digit)
+                    {
+                        digitFlag = true;
+                        int64s.Add(int.Parse(digit.ToString()));
+                        position++;
+                        break;
+                    }
+                }
+                if (digitFlag)
+                    continue;
+                break;
+            }
+
+            for (; position < startIndex + length;)
+            {
+                var trimFlag = false;
+                foreach (var tChar in trimChars)
+                {
+                    if (input[position] == tChar)
+                    {
+                        trimFlag = true;
+                        position++;
+                        break;
+                    }
+                }
+                if (trimFlag)
+                    continue;
+                break;
+            }
+
+            if (position < startIndex + length)
+                return null;
+
+            var val1 = int64s.ToArray<int>();
+            var val = new int[val1.Length];
+            for (int i = 0; i < val1.Length; i++)
+            {
+                val[val1.Length - i - 1] = val1[i]; 
+            }
 
+            value = 0;
+            for (int i = 0; i < val.Length; i++)
+            {
+                try
+                {
+                    if (!minusFlag)
+                    {
+                        checked
+                        {
+                            value = value + val[i] * (long)Math.Pow((double)10, (double)i);
```

Comment:
`Math.Pow` is not precise method for integral types, since it operates with `double`. You could you integral mathematics (plain old `*`).